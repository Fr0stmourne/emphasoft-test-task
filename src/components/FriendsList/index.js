import React from 'react';
import Friend from '../Friend';
import './index.scss';

export default function FriendsList({ friends: friendsResource }) {
  const friends = friendsResource.read();
  return (
    <section className="friends">
      <h2 className="friends__title">Мои друзья:</h2>
      <ul className="friends__list">
        {friends.map((friend, index) => {
          const {
            first_name: firstName,
            last_name: lastName,
            photo_100: photoLink,
            online,
            id,
          } = friend;

          return (
            <li key={index} className="friends__item">
              <Friend
                firstName={firstName}
                lastName={lastName}
                photoLink={photoLink}
                online={online}
                id={id}
              ></Friend>
            </li>
          );
        })}
      </ul>
    </section>
  );
}
