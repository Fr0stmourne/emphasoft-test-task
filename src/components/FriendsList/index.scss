@import '../../scss/variables.scss';

.friends {
  font-family: $primary-font-family;
  display: flex;
  flex-direction: column;
  align-items: center;

  &__list {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  &__item {
    &:not(:last-child) {
      margin-bottom: 20px;
    }
  }
}