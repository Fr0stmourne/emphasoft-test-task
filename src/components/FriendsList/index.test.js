import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import FriendsList from '.';

describe('FriendsList component', () => {
  test('should render the component correctly when no props are provided', () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<FriendsList />);
    expect(tree).toMatchSnapshot();
  });

  test('should render the component with custom props correctly', () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<FriendsList title="My friends:" />);
    expect(tree).toMatchSnapshot();
  });
});
