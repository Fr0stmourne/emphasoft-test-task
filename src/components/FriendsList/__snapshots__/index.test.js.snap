// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendsList component should render the component correctly when no props are provided 1`] = `
<section
  className="friends"
>
  <h2
    className="friends__title"
  >
    Мои друзья:
  </h2>
  <ul
    className="friends__list"
  >
    <li
      className="friends__item"
    >
      <Friend
        firstName="Alex"
        id={1373072}
        lastName="Kouprin"
        online={false}
        photoLink="https://sun9-49.userapi.com/c855536/v855536573/24b6e1/Vx9ANB_8sos.jpg"
      />
    </li>
    <li
      className="friends__item"
    >
      <Friend
        firstName="Константин"
        id={5485951}
        lastName="Константинопольский"
        online={true}
        photoLink="https://sun9-49.userapi.com/c855536/v855536573/24b6e1/Vx9ANB_8sos.jpg"
      />
    </li>
    <li
      className="friends__item"
    >
      <Friend
        firstName="Alexander"
        id={84317567}
        lastName="Bykov"
        online={false}
        photoLink="https://sun9-49.userapi.com/c855536/v855536573/24b6e1/Vx9ANB_8sos.jpg"
      />
    </li>
  </ul>
</section>
`;

exports[`FriendsList component should render the component with custom props correctly 1`] = `
<section
  className="friends"
>
  <h2
    className="friends__title"
  >
    My friends:
  </h2>
  <ul
    className="friends__list"
  >
    <li
      className="friends__item"
    >
      <Friend
        firstName="Alex"
        id={1373072}
        lastName="Kouprin"
        online={false}
        photoLink="https://sun9-49.userapi.com/c855536/v855536573/24b6e1/Vx9ANB_8sos.jpg"
      />
    </li>
    <li
      className="friends__item"
    >
      <Friend
        firstName="Константин"
        id={5485951}
        lastName="Константинопольский"
        online={true}
        photoLink="https://sun9-49.userapi.com/c855536/v855536573/24b6e1/Vx9ANB_8sos.jpg"
      />
    </li>
    <li
      className="friends__item"
    >
      <Friend
        firstName="Alexander"
        id={84317567}
        lastName="Bykov"
        online={false}
        photoLink="https://sun9-49.userapi.com/c855536/v855536573/24b6e1/Vx9ANB_8sos.jpg"
      />
    </li>
  </ul>
</section>
`;
